sockeye/decoder.py

```diff
@@ -375,10 +425,14 @@ def decode(self,
 
         lexical_biases = []
 
-        self.rnn.reset()
+        self.attention_rnn.reset()
         # TODO remove this once mxnet.rnn.SequentialRNNCell.reset() invokes recursive calls on layer cells
-        for cell in self.rnn._cells:
+        for cell in self.attention_rnn._cells:
```

Comment:
can we outsource the whole reset business in a separate private function to make it more concise here?